pkg/openthread/contrib/platform_misc.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) Baptiste Clenet
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Baptiste Clenet <bapclenet@gmail.com>
+ */
+
+
+#include "openthread/types.h"
+#include "openthread/platform/misc.h"
+#include "periph/pm.h"
+
+void otPlatReset(otInstance *aInstance)
+{
+    (void)aInstance;
+    printf("reboot...\n");
+    pm_reboot();
+}
+
+otPlatResetReason otPlatGetResetReason(otInstance *aInstance)
+{
+    (void)aInstance;
+    // TODO: Write me!
```

Comment:
This is the platform abstraction function of the posix example in OpenThread, [as seen here]

[as seen here]:https://github.com/openthread/openthread/blob/fbfd76a990b81f007957e1bd774e51bce742e53e/examples/platforms/posix/misc.c#L40